tests/test_ag3.py

```diff
@@ -1150,3 +1150,45 @@ def test_haplotypes(sample_sets, contig, analysis):
     assert isinstance(d1, xarray.DataArray)
     d2 = ds["call_genotype"].sum(axis=(1, 2))
     assert isinstance(d2, xarray.DataArray)
+
+
+# test v3 sample sets
+def test_sample_cohorts():
+    expected_cols = (
+        "sample_id",
+        "cohort_admin1_year",
+        "cohort_admin1_month",
+        "cohort_admin2_year",
+        "cohort_admin2_month",
+    )
+
+    ag3 = setup_ag3()
+
+    # test v3 wild
+    sample_sets = "v3_wild"
+    df_v3_wild = ag3.sample_cohorts(sample_sets=sample_sets)
+    assert tuple(df_v3_wild.columns) == expected_cols
+    assert len(df_v3_wild) == 2784
+
+    # check v3 is the same
+    sample_sets = "v3"
+    df_v3 = ag3.sample_cohorts(sample_sets=sample_sets)
+    pd.testing.assert_frame_equal(df_v3, df_v3_wild)
+
+    # test a single sample set
+    sample_sets = "AG1000G-UG"
+    df_UG = ag3.sample_cohorts(sample_sets=sample_sets)
+    assert df_UG.sample_id[0] == "AC0007-C"
```

Comment:
Suggest:

```
df_UG_meta = ag3.sample_metadata(sample_sets=sample_set)
assert len(df_UG) == len(df_UG_meta)
assert df_UG['sample_id'].tolist() == df_UG_meta['sample_id'].tolist()
```

Reply:
If compare with sample metadata, could make most of this generic, and use `pytest.parameterize`.